src/flux/task-plan.coffee

```diff
@@ -122,38 +122,66 @@ TaskPlanConfig =
     plan = @_getPlan(id)
     @_change(id, settings: _.extend({}, plan.settings, attributes))
 
-  setDefaultTimes: (course, period) ->
+  setDefaultTimes: (course, period, useCourseDefault) ->
     periodTimes = _.pick(period, 'opens_at', 'due_at')
     {default_open_time, default_due_time} = course
 
-    periodSettings = _.findWhere course.periods, id: period.id
-    {default_open_time, default_due_time} = periodSettings
+    unless useCourseDefault
+      periodSettings = _.findWhere course.periods, id: period.id
+      {default_open_time, default_due_time} = periodSettings
 
     periodTimes.opens_at += " #{default_open_time}" if isDateStringOnly(periodTimes.opens_at)
     periodTimes.due_at += " #{default_due_time}" if isDateStringOnly(periodTimes.due_at)
 
     periodTimes
 
-  setPeriods: (id, courseId, periods, isDefault = false) ->
+  _getDefaultTaskingTimes: (id, courseId, periods, useCourseDefault = true) ->
     plan = @_getPlan(id)
     course = CourseStore.get(courseId)
 
     curTaskings = plan?.tasking_plans
     findTasking = @_findTasking
 
-    tasking_plans = _.map periods, (period) =>
+    _.map periods, (period) =>
       tasking = findTasking(curTaskings, period.id)
-      if not tasking
-        tasking = target_id: period.id, target_type:'period'
+      tasking ?= target_id: period.id, target_type:'period'
+
+      opens_at = period.opens_at or tasking.opens_at
+      due_at = period.due_at or tasking.due_at
+
+      period.opens_at = TimeHelper.makeMoment(opens_at).format(TimeHelper.ISO_DATE_FORMAT) if opens_at?
+      period.due_at = TimeHelper.makeMoment(due_at).format(TimeHelper.ISO_DATE_FORMAT) if due_at?
+
+      periodTimes = @setDefaultTimes(course, period, useCourseDefault) if opens_at? or due_at?
+
```

Comment:
idea for future cleanup:  add a `TimeHelper.toIsoDateString` (or something better named) that you can pass a `string` or `date` to and get back ISO_DATE_FORMAT formatted string.